gnucash/report/reports/standard/category-barchart.scm

```diff
@@ -600,17 +621,23 @@ developing over time"))
 
                 (define (make-cell contents)
                   (gnc:make-html-table-cell/markup "number-cell" contents))
+                  
+                (define (make-cell-percent contents)
+                  (gnc:make-html-table-cell/markup "number-cell" contents " %"))
 
                 (for-each
                  (lambda (date row)
                    (gnc:html-table-append-row!
                     table
-                    (append (list (make-cell date))
-                            (map make-cell row)
-                            (if cols>1?
-                                (list
-                                 (make-cell (apply gnc:monetary+ row)))
-                                '()))))
+                    ( let ((grandt (apply l1norm (map gnc:gnc-monetary-amount row))))
+                     (append (list (make-cell date))
+                             (if ratio-chart? 
+                                 (map make-cell-percent (map (lambda (x) (* (divide-number  (gnc:gnc-monetary-amount x) grandt) 100)) row))
```

Comment:
To be more efficient scheme, don't nest `map` calls; nesting creates intermediate lists which are then immediately consumed by the outer `map` therefore increasing gc workload.

```scheme
(map (lambda (x) (make-cell-percent (* (/ (gnc:gnc-monetary-amount x) grandt) 100) row))
```